download/download_worker.go

```diff
@@ -280,32 +290,33 @@ func findBestMatchingVersion(availibleVers []string, preferredVers string) (stri
 }
 
 // Create the package names from the system information
-func (w *DownloadWorker) formAgentUpgradePackageNames() (string, string, error) {
+func (w *DownloadWorker) formAgentUpgradePackageNames() (*[]string, string, error) {
 	pol, err := persistence.FindNodePolicy(w.db)
 	if err != nil {
-		return "", "", fmt.Errorf("Failed to retrieve node policy from local db: %v", err)
+		return nil, "", fmt.Errorf("Failed to retrieve node policy from local db: %v", err)
 	} else if pol == nil {
-		return "", "", fmt.Errorf("No node policy found in the local db.")
+		return nil, "", fmt.Errorf("No node policy found in the local db.")
 	}
 
-	installTypeProp, err := pol.Properties.GetProperty(externalpolicy.PROP_NODE_OS)
-	if err != nil {
-		return "", "", fmt.Errorf("Failed to find node arch property: err")
+	clusterVers, _ := pol.Properties.GetProperty(externalpolicy.PROP_NODE_K8S_VERSION)
+	if clusterVers.Value != nil {
```

Comment:
Is there any chance that k8s version property is empty for edge cluster?
https://github.com/open-horizon/anax/blob/f895438d8787246425a8b396c8f563b54cb76fce/cutil/cutil.go#L807